Components/SwagImportExport/DbAdapters/ArticlesDbAdapter.php

```diff
@@ -157,23 +165,39 @@ function ($item) {
         } elseif ($filter['productStreamId']) {
             $productStreamId = $filter['productStreamId'][0];
 
+            $contextService = $this->container->get('shopware_storefront.context_service');
+            $streamRepo = $this->container->get('shopware_product_stream.repository');
+            $productNumberSearch = $this->container->get('shopware_search.product_number_search');
+
             /** @var \Shopware\Models\Shop\Repository $shopRepo */
             $shopRepo = $this->modelManager->getRepository(Shop::class);
             $shop = $shopRepo->getActiveDefault();
-            $contextService = Shopware()->Container()->get('shopware_storefront.context_service');
             $context = $contextService->createShopContext($shop->getId());
-            $criteriaService = Shopware()->Container()->get('shopware_search.store_front_criteria_factory');
-            $criteria = $criteriaService->createBaseCriteria([$shop->getCategory()->getId()], $context);
-            $streamRepo = Shopware()->Container()->get('shopware_product_stream.repository');
+            $criteria = new Criteria();
             $streamRepo->prepareCriteria($criteria, $productStreamId);
 
-            $productNumberSearch = Shopware()->Container()->get('shopware_search.product_number_search');
             /** @var ProductNumberSearchResult $products */
             $products = $productNumberSearch->search($criteria, $context);
-            $productNumbers = array_keys($products->getProducts());
+            if (empty($products->getProducts())) {
+                return [];
+            }
 
-            $builder->andWhere('detail.number IN(:productNumbers)')
-                ->setParameter('productNumbers', $productNumbers);
+            if ($filter['variants']) {
+                $articleIds = array_values(array_map(static function (BaseProduct $product) {
```

Comment:
please name the parameter productIds 😬

Reply:
sorry copy/pasted from some lines above. Will replace both instances then 😉